### PR TITLE
Added Floyd Algorithm for searching for duplicates

### PR DESCRIPTION
**Pull request by CyberIkshit**
**Added a cpp file for Floyd Algorithm for cycle detection** Description of Change

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1179"></a>

### CODE REVIEW

**Comment by CyberIkshit** [comment]
Could someone please help me why these automated tests are failing even though my code is perfect?

**Comment by kvedala** [comment]
image

**Comment by CyberIkshit** [comment]
Yeah, I've gone through that but unable to rectify the error from it as it is showing numerous errors in code despite my code is compiling just perfect. Actually, this is my first PR that's why having some difficulty doing it. Hope you'll help.

File: search/FloydAlgorithmForCycleDetection.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+There is only one duplicate number in nums, return this duplicate number in O(1) space and time complexity less than O(n^2) without modifying the original array.
+
+Example-:
+    Input: 5
+           3 1 3 4 2
+    
+    Output: 3
+*/
+#include<bits/stdc++.h>
+using namespace std;
```
Comment (Pardeep009) [inline search/FloydAlgorithmForCycleDetection.cpp:13]: as per repo standards, you cannot use namespaces globally.
using cout, cin etc directly is not allowed, instead use std::cout, std::cout

File: search/FloydAlgorithmForCycleDetection.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    int findDuplicate(vector<int>& nums) {
+        int slow=nums[0];
```
Comment (Pardeep009) [inline search/FloydAlgorithmForCycleDetection.cpp:15]: you have declared c style arrays, use std::array, for example see [https://github.com/TheAlgorithms/C-Plus-Plus/pull/1085](url)

Comment (Pardeep009) [reply search/FloydAlgorithmForCycleDetection.cpp:15]: bits/stdc++ library is not allowed, as it is Linux specific and it also slows down compilation process. so include only required libraries.

**Comment by CyberIkshit** [comment]
Thanks @Pardeep009 for giving such a detailed step by step solution to my problem, it was a great help.

**Comment by CyberIkshit** [comment]
Kindly review and update me if any furthur changes are to be made

**Comment by mishraabhinn** [comment]
> Kindly review and update me if any furthur changes are to be made

Not upto repo standards. 
[Refer](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md)
Thank you.

**Comment by Pardeep009** [comment]
make sure you must add comments for all functions, what they are going to do, what their parameters are, what will be their return type, also add testing for your code. Refer [https://github.com/TheAlgorithms/C-Plus-Plus/pull/1085](url).
if possible, add Wikipedia link.

**Comment by kvedala** [comment]
@Pardeep009 thank you for providing your reviews 😄 
[Contribution guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md)
A complete example of the code structure, writing a good code, the details on find the errors, etc have been elaborated to the greatest extant possible.
Note that "prefect" code, as you refer it, is subjective. The repository goals and objectives and standards might be much more higher and stricter.

**Pull request closed without merging**